DataCapturing/README.md

```diff
@@ -1,313 +1,415 @@
+    func wrapInContext(_ block: (NSManagedObjectContext) throws -> Void) throws             // 2
+    func wrapInContextReturn<T>(_ block: (NSManagedObjectContext) throws -> T) throws -> T  // 3
+    func nextValidIdentifier() throws -> UInt64                                             // 4
+}
+```
 
-#### Setting and Changing Transportation Modes
+It is recommended to have exactly one object of this type per application.
+The recommended way to achive this is to instantiate it as a property to `UIApplication` for *UIKit* applications or as part of the struct adopting the `App` protocol for *SwiftUI*.
+It would probably work to use multiple instances, but this is not guaranteed and unsupported.
+
+**1:** Since the setup of a `DataStoreStack` can be a rather lengthy process, it is put into its own method.
+An application must call this method before any other method from this interface.
+If that is not done, hard to trace and hard to understand errors might occur from within seemingly random places in your application.
+This is especially true if the Cyface SDK default implementation of this interface - which uses *CoreData* - is employed.
+
+**2 and 3:** These two methods realize interaction of application code with persistent code. 
+They provide a context, from which it is safe to work on persistent data objects.
+Data objects may never leave this context.
+Especially if using *CoreData*, managed objects leaving this context, will cause hard to understand errors from seemingly random places in your app.
+One way to avoid this, is to copy all the data required into your own, non managed data structures.
+Be especially careful with forwarding to other thread contexts such as the main thread (i.e. `DispatchQueue.main.async`).
+These are leaving the current context and will lead to errors if they use captured managed objects.
+**4:** Is used as a central entry point to generate storage wide unique identifiers usable by new measurements.
+
+The data is stored following a *CoreData* schema file, which is currently available in Version 13.
+The supported entities are displayed in the following figure:
+
+![Type Diagram showing all the entities used in the Cyface SDK *CoreData* schema.](Documentation.docc/Resources/Cyface-iOS-SDK-Datenbankschema-V13@2x.png)
+
+* **Measurement:** Represent a single measurement in the database. It has two attributes `synchronizable` and `synchronized`, tracking if it is possible and necessary to synchronize this measurement with a storage server. Each measurement also gets a device wide unique `identifier`, which is implemented as a counter.
+* **Track:** A measurement contains one or more tracks. A track is created when the measurement is resumed after a pause.
+* **Event:** This tracks user interactions like putton presses to manipulate an active measurement. Start, stop, pause and resume events are recorded as this entity as well as modality changes.
```

Comment:
typo: **b**utton.... events are recorded as this entity? (did you mean as "an" entity?)

Reply:
No. Since this is an Entity-Relationship-Diagram all the boxes are called "Entity" and thus "Event" is an "Entity" and the four events "Start", "Stop", "Pause" and "Resume" are recorded as an "Event-Entity", so they are recorded as **this** (the explained one) Entity.